LoopKitUI/LoopUIPlugin.swift

```diff
@@ -18,7 +18,10 @@ public protocol ServiceUIPlugin {
     var serviceType: ServiceUI.Type? { get }
 }
 
-public protocol LoopUIPlugin: PumpManagerUIPlugin, CGMManagerUIPlugin {}
+public protocol OnboardingUIPlugin {
+    var onboardingType: OnboardingUI.Type? { get }
+}
 
-// TODO: Remove LoopUIPlugin after updating OmniKitPlugin and MinimedKitPlugin in rileylink_ios to explicitly
-// use PumpManagerUIPlugin and/or CGMManagerUIPlugin rather than LoopUIPlugin
+public protocol SupportUIExtension {
```

Comment:
Hmm, you are right in that it isn't technically needed. However, if we did away with the `SupportUIExtension` layer then the plugin framework dynamic class (e.g. `TidepoolSupportPlugin`) would have to directly conform to `SupportUI` and we'd have lost flexibility in how the plugin implements it.

Reply:
I guess I think that "Support" shouldn't be a new type of plugin. It should just be an optional thing that existing plugins can provide.

Reply:
People might need support for a pump, or a cgm, or a service, so I think it should be associated with those.

Reply:
What if it is support that is *not* related to a pump, cgm, or service, like our current Support menu item for submitting an Adverse Event Report?

Reply:
Replying from email doesn't apparently make it go in the thread properly. Re posting here: 

Wouldn’t that fit into the TidepoolService plugin? How is the adverse event going to be submitted? Via the Tidepool service, no?

Reply:
Currently, the support menu item for both LoopSupport and TidepoolSupport simply opens a URL in the browser, . For TidepoolSupport, some of the data being collected is pre-populated (via URL parameters), but there is a large web-based form that can be completed. The form is hosted on support.tidepool.org (via ZenDesk, I believe).

Reply:
That still makes some sense to put in TidepoolService plugin, no?

Reply:
Just so I'm clear, are you proposing that I make `TidepoolService` implementation conform to the `SupportUI` protocol, or just move the `TidepoolSupport` implementation into the `TidepoolService` plugin? And, in either case, remove the entire `TidepoolSupport` plugin?

Reply:
@ps2 We still need resolution on this thread. How do we want to proceed here?

Reply:
Either make TidepoolServiceUI conform to SupportUI, or make the ServiceUI be able to vend a SupportUI object, is what I was thinking. And yes, I don't think we'd need a separate plugin then.

Reply:
How does this work for NightscoutService? As it is optional (e.g. the user can choose "Setup **without** Nightscout", there may not be an instance of the NightscoutService after onboarding.

Reply:
Hmm, I suppose we could have extensions vend a support too. This seems a bit a of a special case. I do want to have DIY offer some kind of support (like to file github tickets, or maybe at some point upload issue reports). Either having extensions vend this as well, or DIY can make a service (or reuse NS service) that doesn't actually require login.

Reply:
The current implementations does have the extension vend this. 

Given your concerns, I think the best thing to do here is to simply rename `SupportUIExtension` to `SupportUIPlugin`, add a `LoopPluginBundleKey.supportIdentifier` and have the PluginManager find `SupportUI` implementations the same way as it does CGM managers, pump managers, and services. This way it gives the plugin full control over how it provides `SupportUI`. It can attach it to a service, add it to a plugin extension, or whatever.